webapp/src/components/tutorial/TutorialContainer.tsx

```diff
@@ -110,17 +110,14 @@ export function TutorialContainer(props: TutorialContainerProps) {
     const hintMarkdown = steps[visibleStep].hintContentMd;
 
     const validateTutorialStep = async () => {
-        const globalValidators =
-            tutorialOptions.globalValidationConfig?.validators ??
-            PopulateValidatorCache(tutorialOptions.globalValidationConfig);
-        const localValidators =
-            currentStepInfo.localValidationConfig?.validators ??
-            PopulateValidatorCache(currentStepInfo.localValidationConfig);
-
-        let validators: pxt.Map<CodeValidator> = {
-            ...(globalValidators ?? {}),
-            ...(localValidators ?? {})
-        };
+        let validators: pxt.Map<CodeValidator> = {};
+
+        currentStepInfo.localValidationConfig?.validatorsMetadata?.forEach(v => validators[v.validatorType] = GetValidator(v));
+        tutorialOptions.globalValidationConfig?.validatorsMetadata?.forEach(v => {
+            if(!validators[v.validatorType]) {
```

Comment:
```suggestion
            if (!validators[v.validatorType]) {
```
nit

Reply:
One day I'll remember to check for these before sending the PR :)

Reply:
I mean really we should just automatically run prettier or something so we never have to think about formatting, just have to find a way to get prettier to ignore our hacky `lf` logic (or support lfs going to new lines in the cli). I'll try to find some free time to do it someday haha